### ISSUE TITLE
Suggestions on UX

### ISSUE THREAD

**Issue by khashayarxy** [LorenEteval/Furious#45]
Hey there, I hope you are well.
First of all, Thank you for making an amazing app for us.

There are some things that I think should be changed for a better user experience.

1) The app should always open in a window and not minimized. (It's hard for some people to understand that the app has opened.)
although a minimized opening is useful for startup.

2) After updating the subscription, it gets connected to the first server, which is always a time/date server and not an actual server for proxying.
So either it should not change the selected server in the first place or connect to the last server on the list (v2rayNG does that).

3) For some people, having other VPNs installed makes the app not work and show an error.

4) Adding an option to connect with Right-Click on the server and selecting Connect would be nice.

5) Adding an option to change the theme to dark mode would be appreciated. (a simple black theme like Github would be fine.)

6) I can help with translation to Persian (Farsi) if you want. (since most people from Iran use it.)

7) The editor section should always be hidden and only shown when it's wanted. (Most normal users wouldn't understand it.)

8) Showing emojis would be another nice thing to add. (flag emojis and that kind of thing)

9) Releasing an App Store version of the app is necessary. (Most people with macOS don't trust third-party apps.)

10) Last but not least, for me, routing settings don't work even with the latest release, 0.2.13.
When I activated any routing beside Bypass China, some websites wouldn't load.
Also, Iranian websites that have geo-blocks don't get opened with Bypass Iran routing.

**Comment by LorenEteval** [comment]
Thank you and thanks for the advice (that's a lot). I'll just reveal some implementation details using this chance. Free to ignore them.

1. I may have to refer to the behavior of other clients for this. Initially my UI design was inspired by v2rayN (if anyone noticed).

2. Will be added. Need a little trick to do this. Current `update subs` logic: delete then re-import:

https://github.com/LorenEteval/Furious/blob/83ec031e04c3c7a86c8083adb84ecb7f39969aad/Furious/Widget/EditConfiguration.py#L801-L812

That's why it goes to the first because the rest are deleted asynchronously at first place.

3. This confuses me. What is the reason for this?

4. Will be added

5. Currently the theme follows the system. A standalone light/dark mode requires my investigation.

6. It would be nice to have this. Much appreciate it. But not now since I'll refactor some of translation logic to make it more human-friendly in 0.3.x (it will be a complete standalone translation system independent with Qt).

7. Unfortunately right now I don't have any other better ideas for this. The editor may still be remained in 0.3.x. My plan is to also make it possible to change configuration via a pop-up UI window. (no difficulty but quite a lot work in the Qt designer).

8. Doable. I'll consider support it.

9. Need my investigation. I'll consider support it.

10. Thanks for the feedback, however I need more bug report for this. I assume you are referring to Xray-core and I checked the implementation and it seems right to me.

The Xray-core asset environment variable is added when application starts up. It points to the data dir in Furious application location (to ensure that assets can be found and loaded).

https://github.com/LorenEteval/Furious/blob/83ec031e04c3c7a86c8083adb84ecb7f39969aad/Furious/Widget/Application.py#L249-L252

https://github.com/LorenEteval/Furious/blob/83ec031e04c3c7a86c8083adb84ecb7f39969aad/Furious/Widget/Application.py#L260

```json
{
  "domainStrategy": "IPIfNonMatch",
  "domainMatcher": "hybrid",
  "rules": [
    {
      "type": "field",
      "domain": [
        "geosite:category-ads-all",
        "iran:ads"
      ],
      "outboundTag": "block"
    },
    {
      "type": "field",
      "domain": [
        "iran:ir",
        "iran:other"
      ],
      "outboundTag": "direct"
    },
    {
      "type": "field",
      "ip": [
        "geoip:private",
        "geoip:ir"
      ],
      "outboundTag": "direct"
    },
    {
      "type": "field",
      "port": "0-65535",
      "outboundTag": "proxy"
    }
  ]
}
```

This RoutingObject gets loaded in the user configuration in `Bypass Iran` mode

---

0.3.x won't be released until next year. ~~This project has consumed a lot of my time and energy. I need time to focus on my real life.~~ It's always a privilege and motivation to see people like it, and see it helps people bypass GFW.

**Comment by khashayarxy** [comment]
No problem, and thank you. I appreciate your response.

1) That's alright. That feedback comes from my clients. I have "Startup on Boot" enabled and didn't notice it much.

2) I understand. Thanks for changing it.

3) It shows an error about the X-ray core.

I have no idea why it happened, but some of the things that I did to fix it are as follows:
-Deleting any other VPNs and VPN plugins.
-Using "ipconfig /flushdns" in cmd.
-Removing any proxy data in Internet Options > Connections > Lan Settings > Proxy Server and Advanced section.

4) That would be great; thank you.

5) I have Windows 11 and Dark Mode enabled, but the app theme doesn't change.

6) I'm always happy to help with that. You can just send me the phrases, and I will send you translations. Also, you can use DeepL for near-accurate translations for other languages.

7) A pop-up would be better than what it currently is, if it stays closed after closing.

8) Thank you.

9) I appreciate that.

10) I have no clue about the reason, but the geo:ir part should open websites with the ir domain and geo-block, right?
for "irancell.ir" and "mci.ir", I tried it, and it didn't work. So probably something is not working correctly. I may be wrong about the purpose of this feature.

Again, thank you so much for looking into these things. This app is currently the best for Windows and Linux, and when launched on the macOS App Store, it would be anyone's go-to app. Thanks for considering my suggestions, and have a great time while resting :)

**Comment by khashayarxy** [comment]
Here are some other thoughts for future updates:

11. Auto-update can be useful. Lots of people forget to update their subscription.
It can be done every time user opens the app and also with a default timer set to every 12 hour.

12. Importing configs and subscriptions could be easier.
For example: user copies subscription link and goes to "Subscription" menu and selects "import subscription" and it automatically adds the subscription with a default name and an auto-update with a 12H timer.

13. Since there is an option for exporting QR Code, importing from QR Code would be great.

14. The option for connecting with Right-Click can be 2 things on the menu:
Select this Server
Connect to this Server (which selects and connects at the same time)

Also, you should consider putting crypto addresses for donating. Some popular ones are TRX, BNB, TON, ETH, BTC and USDT.
You can use [nowpayments.io](https://nowpayments.io/) which offers many crypto options.

**Comment by LorenEteval** [comment]
Thanks. Some of these are easy to improve. I'll refactor the code base before we can move to 0.3.

For 10 I got similar report #19 , but no futher confirmation (also just one report). Since I'm not in Iran, I cannot test if `Bypass Iran` mode works so it's hard for me to locate any potential bugs.

0.3.x will be a fully polished version.

**Comment by khashayarxy** [comment]
Just wanted to add another suggestion here.

15. If the app could actively show the status of internet connection to the user, it would clear the confusion when the internet is unstable and the user doesn't know if there's an issue with the VPN or the internet.

Could be shown like this: 
Internet Status: 🔴 Offline
Internet Status: 🟢 Online

**Comment by LorenEteval** [comment]
Closed this temporarily to focus on my priorities. I'll reopen this when we are ready to make changes. Thank you for the advice
